### PR TITLE
feat: GH workflow for building Envoy on Linux

### PR DESCRIPTION
**Pull request by michaelbeaumont**

### CODE REVIEW

File: .github/workflows/build.yaml [hunk 1]
```diff
@@ -0,0 +1,37 @@
...
+  build:
+    runs-on: ubuntu-latest
+    defaults:
+      run:
+        working-directory: terraform
+    steps:
+      - uses: actions/checkout@v3
+      - run:
+          ssh-keygen -t ed25519 -q -N "" -f ~/.ssh/id_ed25519
+      - uses: hashicorp/setup-terraform@v2
```
Comment (lahabana) [inline .github/workflows/build.yaml:18]: pin the terraform version (in the past it used to break often)

File: .github/workflows/build.yaml [hunk 2]
```diff
@@ -0,0 +1,37 @@
...
+        run: terraform output --raw public_ip
+      - name: Build Envoy
+        uses: appleboy/ssh-action@v0.1.7
+        with:
+          host: ${{ steps.public-ip.outputs.stdout }}
+          username: admin
+          key_path: ~/.ssh/id_ed25519
+          script: |
+            git clone https://github.com/kumahq/kuma && cd kuma
+            BUILD_ENVOY_FROM_SOURCES=true make build/envoy
```
Comment (lahabana) [inline .github/workflows/build.yaml:37]: Need terraform destroy

File: terraform/main.tf [hunk 3]
```diff
@@ -0,0 +1,123 @@
...
+
+sudo mkdir -m 0755 -p /etc/apt/keyrings
+curl -fsSL https://download.docker.com/linux/debian/gpg | sudo gpg --dearmor -o /etc/apt/keyrings/docker.gpg
+
+echo \
+  "deb [arch=$(dpkg --print-architecture) signed-by=/etc/apt/keyrings/docker.gpg] https://download.docker.com/linux/debian \
+  $(lsb_release -cs) stable" | sudo tee /etc/apt/sources.list.d/docker.list > /dev/null
+
+apt-get update
+apt-get install -y docker-ce docker-ce-cli containerd.io docker-buildx-plugin docker-compose-plugin
```
Comment (lahabana) [inline terraform/main.tf:84]: You might save yourself some time by having an action to create an image with all this already installed. I've user https://www.packer.io/ in the past and it's very easy

Comment (michaelbeaumont) [reply terraform/main.tf:84]: I'd like to do this in a separate PR. It doesn't take much time to run the user data at start up and it's easier to iterate on.

**Comment by michaelbeaumont** [comment]
@lahabana https://github.com/michaelbeaumont/envoy-builds/actions/runs/4312283163

File: .github/workflows/build.yaml [hunk 4]
```diff
@@ -0,0 +1,70 @@
...
+        run: |
+          while ! ssh -o StrictHostKeyChecking=no admin@${{ steps.public-ip.outputs.stdout }} 'test -e ready'
+          do
+            sleep 5
+          done
+      - name: Build Envoy
+        run: |
+          ssh -o StrictHostKeyChecking=no admin@${{ steps.public-ip.outputs.stdout }} <<'EOF'
+            git clone https://github.com/kumahq/kuma && cd kuma
+            BUILD_ENVOY_FROM_SOURCES=true make build/envoy
```
Comment (lahabana) [inline .github/workflows/build.yaml:59]: Can we just have these files here and use these? So we can get rid of this code in Kuma in the future?

**Review by lahabana — APPROVED** [review]
This makes a lot of sense. My only point is relying on scripts in kumahq/kuma but that can always be fixed later

File: .github/workflows/build.yaml (hunk 4 above)
Comment (michaelbeaumont) [reply .github/workflows/build.yaml:59]: Yes, definitely. My next steps are PR for darwin and then moving the build stuff to this repo.

**Pull request merged**